File: IMS/src/main/java/com/revature/services/SupplierService.java
package com.revature.services;

import java.util.List;
import java.util.Set;

import com.revature.beans.Supplier;

public interface SupplierService {
	public int addSupplier(Supplier sup);
	public Supplier getSupplierById(int id);
	public Set<Supplier> getSuppliers();
	public List<Supplier> getSuppliersCriteria();
	public void updateSupplier(Supplier sup);	
	public void deleteSupplier(Supplier sup);
	public void deleteSupplierById(int id);
}

File: client-side/src/app/supplier/supplier.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { CoreService } from '../core/core.service';
import { Observable, pipe } from 'rxjs';
import { map } from 'rxjs/operators';
import { Supplier } from '../class/supplier';
import { Item } from '../class/Item';

@Injectable({
    providedIn: 'root'
  })

  export class SupplierService {
    private appUrl = this.coreService.getURL() + '/supplier';
    private headers = this.coreService.getHeader();

    constructor(private http: HttpClient, private coreService: CoreService) {}

    getSuppliers(): Observable<Supplier[]> {
        return this.http
          .get(this.appUrl + '/all')
          .pipe(map(resp => resp as Supplier[]));
      }
    addSupplier(sup: Supplier): Observable<Supplier>{
        const body = {};
        return this.http
            .post(this.appUrl, sup,{ headers: this.headers})
            .pipe(map(resp => resp as Supplier))
    }
    updateSupplier(sup: Supplier): Observable<Supplier>{
        return this.http
            .put(this.appUrl, sup,{headers: this.headers})
            .pipe(map(resp => resp as Supplier))
    }
    deleteSupplier(id: Number): Observable<Supplier>{
        return this.http
            .delete(this.appUrl+'/'+id, {headers: this.headers})
            .pipe(map(resp => resp as Supplier))
    }
}